data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -225,18 +257,19 @@ private CloseableIterable<T> applyPosDeletes(CloseableIterable<T> records) {
     List<CloseableIterable<Record>> deletes = Lists.transform(posDeletes, this::openPosDeletes);
 
     // if there are fewer deletes than a reasonable number to keep in memory, use a set
-    if (posDeletes.stream().mapToLong(DeleteFile::recordCount).sum() < setFilterThreshold) {
-      return Deletes.filter(records, this::pos, Deletes.toPositionIndex(filePath, deletes));
+    if (posDeletes.stream().mapToLong(DeleteFile::recordCount).sum() < streamFilterThreshold) {
+      return Deletes.filter(records, this::pos, Deletes.toPositionIndex(filePath, deletes, counter));
     }
 
-    return Deletes.streamingFilter(records, this::pos, Deletes.deletePositions(filePath, deletes));
+    return Deletes.streamingFilter(records, this::pos, Deletes.deletePositions(filePath, deletes), counter);
```

Comment:
This is a code path that was not previously exercised by the unit tests.
We add a configuration that allows us to control the `streamFilterThreshold` and cause this code path to be taken.